tangelo/problem_decomposition/dmet/dmet_problem_decomposition.py

```diff
@@ -115,7 +115,7 @@ def __init__(self, opt_dict):
         # Converting our interface to pyscf.mol.gto and pyscf.scf (used by this
         # code).
         self.mean_field = self.molecule.mean_field
-        self.molecule = self.molecule.to_pyscf(self.molecule.basis)
+        self.molecule = self.molecule.solver.to_pyscf(self.molecule, self.molecule.basis)
```

Comment:
I moved `to_pyscf` to outside the solver class. But Molecule still has a IntegralSolver attached to it. IntegralSolver does not have direct access to it so all IntegralSolver functions require molecule to pass itself to solver. We could change all the IntegralSolver to return all variables instead of modifying sqmol but that list may get long.